src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs

```diff
@@ -2430,6 +2400,118 @@ public async Task<JArray> GetObjectValues(int objectId, GetObjectCommandOptions
                 ret = retAfterRemove;
             }
             return ret;
+
+            async Task<JArray> GetFieldsValue(List<FieldTypeClass> fields, bool isFirstElement, bool isRootHidden = false, bool isArray = false)
+            {
+                if (getCommandType.HasFlag(GetObjectCommandOptions.ForDebuggerProxyAttribute))
+                {
+                    fields = fields.Where(field => field.IsPublic).ToList();
+                }
+
+                using var commandParamsWriter = new MonoBinaryWriter();
+                commandParamsWriter.Write(objectId);
+                commandParamsWriter.Write(fields.Count);
+                foreach (var field in fields)
+                {
+                    commandParamsWriter.Write(field.Id);
+                }
+
+                var retDebuggerCmdReader = await SendDebuggerAgentCommand(CmdObject.RefGetValues, commandParamsWriter, token);
+
+                JArray objFields = new JArray();
+                foreach (var field in fields)
+                {
+                    long initialPos = retDebuggerCmdReader.BaseStream.Position;
+                    int valtype = retDebuggerCmdReader.ReadByte();
+                    retDebuggerCmdReader.BaseStream.Position = initialPos;
+                    var fieldValue = await CreateJObjectForVariableValue(retDebuggerCmdReader, field.Name, isFirstElement, field.TypeId, getCommandType.HasFlag(GetObjectCommandOptions.ForDebuggerDisplayAttribute), token);
+                    if (ret.Where(attribute => attribute["name"].Value<string>().Equals(fieldValue["name"].Value<string>())).Any())
+                    {
+                        return null;
```

Comment:
Shouldn't this continue to the next field? If so, and it didn't break any tests, then we need a new test for this.

Reply:
You are right about continuing here. If we already have an element with a certain name in objectValues then we shouldn't add another one of the same name. But I'm not sure how we can test it. Creating nested objects of the same names does not result in entering this condition. I'm not sure what situation the author meant originally for null return. It was added in 4f4f0dbd5e674f4d19d8d51fd6717b9316224852, @thaystg can you help me here?

Reply:
Originally there was continue and I changed it. Test InspectTypeInheritedMembers with ClassGetPropertiesTestData checks it.